src/Kiota.Builder/CodeDOM/CodeFile.cs

```diff
@@ -6,6 +6,12 @@ namespace Kiota.Builder.CodeDOM;
 
 public class CodeFile : CodeBlock<CodeFileDeclaration, CodeFileBlockEnd>
 {
+    public IEnumerable<CodeNamespace> Namespaces => InnerChildElements.Values.OfType<CodeNamespace>();
```

Comment:
additionally, we should probably have some ordering by name to get a deterministic output